Jenkinsfile

```diff
@@ -13,177 +13,158 @@
-          withCredentials([usernameColonPassword(credentialsId: 'jenkins-x-chartmuseum', variable: 'CHARTMUSEUM_AUTH')]) {
-            script {
-              // initialize Helm and package chart
-              sh """
-                helm init --client-only
-
-                helm repo add elastic https://helm.elastic.co/
-                helm repo add kubernetes-charts https://kubernetes-charts.storage.googleapis.com/
-                helm repo add bitnami https://charts.bitnami.com/bitnami
-
-                helm dependency update ${CHART_NAME}
+            currentBuild.description = "${BUILD_VERSION}"
+            updateVersion("${BUILD_VERSION}")
 
-                helm package ${CHART_NAME}
-              """
+            echo "Package chart version: ${BUILD_VERSION}"
+            sh "helm3 package ${CHART_NAME}"
 
-              // install the chart into a test namespace that will be cleaned up afterwards
-              sh """
-                jx step helm install ${CHART_NAME} \
-                  --name=${TEST_RELEASE} \
-                  --namespace=${TEST_NAMESPACE} \
-                  --set=nuxeo.image.tag=11.x # TODO remove when NXP-28504 is done and the latest tag (default) is available
-              """
-
-              // check deployment status, exits if not OK
-              sh """
-                kubectl rollout status deployment ${TEST_K8S_RESSOURCE} \
-                  --namespace=${TEST_NAMESPACE} \
-                  --timeout=${TEST_ROLLOUT_STATUS_TIMEOUT}
-              """
-
-              // check running status
-              def runningStatus = sh(returnStatus: true, script: "./running-status.sh http://${TEST_SERVICE_DOMAIN}/nuxeo")
-              if (runningStatus != 0) {
-                currentBuild.result = 'FAILURE';
-                currentBuild.description = "Running status is not OK."
-                error(currentBuild.description)
-              }
+            echo 'Test chart'
+            // install the chart into a test namespace that will be cleaned up afterwards
+            sh """
+              kubectl create namespace ${TEST_NAMESPACE}
+              helm3 install ${TEST_RELEASE} ${CHART_NAME} \
+                --namespace=${TEST_NAMESPACE} \
+                --set=image.tag=11.x
+            """
+            // check deployment status, exit if not OK
+            sh """
+              kubectl rollout status deployment ${TEST_K8S_RESSOURCE} \
+                --namespace=${TEST_NAMESPACE} \
+                --timeout=${TEST_ROLLOUT_STATUS_TIMEOUT}
+            """
+            // check running status
+            def runningStatus = sh(returnStatus: true, script: "test/running-status.sh http://${TEST_SERVICE_DOMAIN}/nuxeo")
+            if (runningStatus != 0) {
+              currentBuild.result = 'FAILURE'
+              currentBuild.description = "Running status is not OK."
+              error(currentBuild.description)
+            }
```

Comment:
It helps to wait for the pod condition being "ready" first, then the curl request is only informative.
Here's an alternate method to wait for Nuxeo and debug in case of failure:

```
/**
 * Wait for Nuxeo Kubernetes application's deployment, then wait for the pod being effectively ready
 * and finally check the running status.
 * In case of error, debug information is logged.
 * @param name Nuxeo app name
 * @param url Nuxeo URL
 */
def void waitForNuxeo(String name, String url) {
    script {
        try {
            sh "kubectl -n ${PREVIEW_NAMESPACE} rollout status deployment $name"
            sh "kubectl -n ${PREVIEW_NAMESPACE} wait --for=condition=ready pod -l app=$name --timeout=-0"
            sh "curl --retry 10 -fsSL $url/nuxeo/runningstatus"
        } catch (e) {
            sh "kubectl -n ${PREVIEW_NAMESPACE} get all,configmaps,endpoints,ingresses"
            sh "kubectl -n ${PREVIEW_NAMESPACE} describe pod --selector=app=$name"
            sh "kubectl -n ${PREVIEW_NAMESPACE} logs --selector=app=$name --all-containers --tail=1000"
            throw e
        }
    }
}
```
then `waitForNuxeo("preview", AI_CLOUD_PUBLIC_URL)`

https://github.com/nuxeo/nuxeo-ai-integration/blob/master/Jenkinsfile#L120-L140

Reply:
Thanks, interesting.
Wondering what the `wait` does compared to the `rollout status`. Have you observed cases where the  deployment is updated successfully (`rollout status` succeeds) but the pod is not ready?

I've added the debug info :+1:.

Reply:
> Have you observed cases where the deployment is updated successfully (rollout status succeeds) but the pod is not ready?
Yes, precisely. The pod readiness is not part of the condition for the rollout status being true.

Contrariwise, the `curl` call to Nuxeo running status **is** redundant with the `wait` for readiness and can likely be skipped. I wanted to do that final check though.